docs/src/pages/demos/breadcrumbs/RouterBreadcrumbs.tsx

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable no-nested-ternary */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles, WithStyles, Theme, createStyles } from '@material-ui/core/styles';
+import List from '@material-ui/core/List';
+import Link, { LinkProps } from '@material-ui/core/Link';
+import NoSsr from '@material-ui/core/NoSsr';
+import ListItem from '@material-ui/core/ListItem';
+import Collapse from '@material-ui/core/Collapse';
+import ListItemText from '@material-ui/core/ListItemText';
+import Typography from '@material-ui/core/Typography';
+import ExpandLess from '@material-ui/icons/ExpandLess';
+import ExpandMore from '@material-ui/icons/ExpandMore';
+import Breadcrumbs from '@material-ui/core/Breadcrumbs';
+import { Route, MemoryRouter } from 'react-router';
+import { Link as RouterLink, LinkProps as RouterLinkProps } from 'react-router-dom';
+
+interface RouterBreadcrumbsState {
+  readonly open: boolean;
+}
+
+interface ListItemLinkProps extends LinkProps {
+  to: string;
+  open?: boolean;
+}
+
+interface RouterBreadcrumbsProp extends WithStyles<typeof styles> {}
+
+const breadcrumbNameMap: { [key: string]: string } = {
+  '/inbox': 'Inbox',
+  '/inbox/important': 'Important',
+  '/trash': 'Trash',
+  '/spam': 'Spam',
+  '/drafts': 'Drafts',
+};
+
+function ListItemLink(props: ListItemLinkProps) {
+  const { to, open, ...other } = props;
+  const primary = breadcrumbNameMap[to];
+
+  return (
+    <li>
+      <ListItem button component={RouterLink} to={to} {...other}>
+        <ListItemText primary={primary} />
+        {open != null ? open ? <ExpandLess /> : <ExpandMore /> : null}
+      </ListItem>
+    </li>
+  );
+}
+
+ListItemLink.propTypes = {
+  open: PropTypes.bool,
+  to: PropTypes.string.isRequired,
+};
+
+const styles = (theme: Theme) =>
+  createStyles({
+    root: {
+      display: 'flex',
+      flexDirection: 'column',
+      width: 360,
+    },
+    lists: {
+      backgroundColor: theme.palette.background.paper,
+      marginTop: theme.spacing(1),
+    },
+    nested: {
+      paddingLeft: theme.spacing(4),
+    },
+  });
+
+interface LinkRouterProps extends LinkProps {
+  to: string;
+  replace?: boolean;
+}
+
+const LinkRouter = (props: LinkRouterProps) => <Link {...props} component={RouterLink as any} />;
+
+class RouterBreadcrumbs extends React.Component<RouterBreadcrumbsProp, RouterBreadcrumbsState> {
+  state = {
+    open: true,
+  };
+
+  handleClick = () => {
+    this.setState(state => ({ open: !state.open }));
+  };
+  static propTypes: any;
```

Comment:
Is this needed?

Reply:
if i remove , I got type error 
`Property 'propTypes' does not exist on type 'typeof RouterBreadcrumbs'.`

Reply:
I tried to write this way, I will not report type error.
```typescript
(RouterBreadcrumbs as React.ComponentClass<RouterBreadcrumbsProp, RouterBreadcrumbsState>).propTypes = {
  classes: PropTypes.object.isRequired,
} as any;
```

Reply:
@Adherentman The pattern you used in the comment is the one we use in the other demos as well. We should be consistent.

Reply:
@eps1lon ok，Can I improve this demo by changing it to hooks next time?

Reply:
Sure. With hooks we don't have that issue at all. Conversion to hooks should be done in a separate PR though.

Reply:
I change this agin.but I will change this to hooks in the near future.